src/_pytest/python.py

```diff
@@ -152,7 +152,10 @@ def pytest_configure(config):
 def pytest_pyfunc_call(pyfuncitem):
     testfunction = pyfuncitem.obj
     iscoroutinefunction = getattr(inspect, "iscoroutinefunction", None)
-    if iscoroutinefunction is not None and iscoroutinefunction(testfunction):
+    isasyncgenfunction = getattr(inspect, "isasyncgenfunction", None)
+    if (iscoroutinefunction is not None and iscoroutinefunction(testfunction)) or (
+        isasyncgenfunction is not None and isasyncgenfunction(testfunction)
+    ):
         msg = "Coroutine functions are not natively supported and have been skipped.\n"
```

Comment:
Maybe adapt the message to also mention async generator functions?

I'd actually suggest changing "Couroutine function" with "async function" and then it can sort of work for both, and is more familiar than "coroutine".

Reply:
I've gone for `async def`

Reply:
https://github.com/python/cpython/blob/98ef920/Doc/library/asyncio-task.rst#L146